staking_deposit/key_handling/key_derivation/mnemonic.py

```diff
@@ -90,30 +92,48 @@ def _get_checksum(entropy: bytes) -> int:
     return int.from_bytes(SHA256(entropy), 'big') >> (256 - checksum_length)
 
 
-def verify_mnemonic(mnemonic: str, words_path: str) -> bool:
+def abbreviate_words(words: Sequence[str]) -> List[str]:
     """
-    Given a mnemonic, verify it against its own checksum."
+    Given a series of word strings, return the 4-letter version of each word (which is unique according to BIP39)
+    """
+    return [normalize('NFKC', word)[:4] for word in words]
+
+
+def reconstruct_mnemonic(mnemonic: str, words_path: str) -> Optional[str]:
+    """
+    Given a mnemonic, a reconstructed the full version (incase the abbreviated words were used)
+    then verify it against its own checksum
     """
     try:
         languages = determine_mnemonic_language(mnemonic, words_path)
     except ValueError:
-        return False
+        return None
+    reconstructed_mnemonic = None
     for language in languages:
         try:
-            word_list = _get_word_list(language, words_path)
-            mnemonic_list = mnemonic.split(' ')
+            word_list = abbreviate_words(_get_word_list(language, words_path))
+            mnemonic_list = abbreviate_words(mnemonic.lower().split(' '))
```

Comment:
what do you think about making it more clear that they are abbreviated:
- `word_list` -> `abbrev_word_list`
- `mnemonic_list` -> `abbrev_mnemonic_list`

Reply:
Sure, I'll add a PR